elastic/integrations
# [Kubernetes][Container Logs] Add ability to configure `data_stream.dataset` & "Custom configuration"

Enhancement

## What does this PR do?



This PR does 2 things for the Kubernetes Container Log integration:

1. Adds the ability to configure `data_stream.dataset`.
2. Adds the ability to add custom configurations to container logs

Both changes are intended to bring this integration closer inline with the [Custom Log](https://github.com/elastic/integrations/tree/main/packages/log) integration.

## Checklist


## Author's Checklist


- `data_stream.dataset` is now configurable
- "Custom configuration" section injects additional yaml configuration.

<details>
<summary>Before Upgrade</summary>

```yaml
inputs:
  - id: filestream-container-logs-329e9d5f-dd99-438e-8271-3e3598c8300a
    name: kubernetes-1
    revision: 1
    type: filestream
    use_output: default
    meta:
      package:
        name: kubernetes
        version: 1.32.0
    data_stream:
      namespace: default
    package_policy_id: 329e9d5f-dd99-438e-8271-3e3598c8300a
    streams:
      - id: >-
          kubernetes-container-logs-${kubernetes.pod.name}-${kubernetes.container.id}
        data_stream:
          dataset: kubernetes.container_logs
          type: logs
        paths:
          - '/var/log/containers/*${kubernetes.container.id}.log'
        prospector.scanner.symlinks: true
        parsers:
          - container:
              stream: all
              format: auto
```

</details>

<details>
<summary>After Upgrade (No modifications)</summary>

```yaml
inputs:
  - id: filestream-container-logs-329e9d5f-dd99-438e-8271-3e3598c8300a
    name: kubernetes-1
    revision: 2
    type: filestream
    use_output: default
    meta:
      package:
        name: kubernetes
        version: 1.33.0
    data_stream:
      namespace: default
    package_policy_id: 329e9d5f-dd99-438e-8271-3e3598c8300a
    streams:
      - id: >-
          kubernetes-container-logs-${kubernetes.pod.name}-${kubernetes.container.id}
        data_stream:
          dataset: kubernetes.container_logs
        prospector.scanner.symlinks: true
        paths:
          - '/var/log/containers/*${kubernetes.container.id}.log'
        parsers:
          - container:
              stream: all
              format: auto
```

</details>

**Note:** After the upgrade, the policy configuration no longer is setting `streams.data_stream.type: logs`. I don't believe this matters as I check the Custom logs integration and it does the same thing, as well as the static tests passed. But if desired, I can have it add `streams.data_stream.type: logs`.

<details>
<summary>After Upgrade (With modifications)</summary>

```yaml
inputs:
  - id: filestream-container-logs-329e9d5f-dd99-438e-8271-3e3598c8300a
    name: kubernetes-1
    revision: 4
    type: filestream
    use_output: default
    meta:
      package:
        name: kubernetes
        version: 1.33.0
    data_stream:
      namespace: default
    package_policy_id: 329e9d5f-dd99-438e-8271-3e3598c8300a
    streams:
      - id: >-
          kubernetes-container-logs-${kubernetes.pod.name}-${kubernetes.container.id}
        data_stream:
          dataset: custom.dataset
        prospector.scanner.symlinks: true
        paths:
          - '/var/log/containers/*${kubernetes.container.id}.log'
        file_identity.inode_marker.path: /logs/.filebeat-marker #<-- Custom configuration line here.
        parsers:
          - container:
              stream: all
              format: auto
```

</details>

## How to test this PR locally



1. Install current integration
2. Upgrade integration to this PR
3. Observe that policy is similar to before the upgrade and the integration works
4. Change `data_stream.dataset` value, and observe change is reflected in agent policy
5. Change Custom configuration value, and observe change is reflected in agent policy

## Related issues


- Closes #5551

## Screenshots



![image](https://user-images.githubusercontent.com/8277432/225468237-ecf17208-53e3-4b74-8d54-9967cfde9227.png)
